// elegant-config/index.js
module.exports = {
    SERVER: {
        PORT: 9090,
        POST_MAX_SIZE: 40, //MB
        UPLOAD_MAX_FILE_SIZE: 40 //MB , less than or equal POST_MAX_SIZE
    },
    DIR: {
        PROJECT: __dirname, // project directory
        SESSION: ".session-data", // session data
        INTERCEPTORS: 'app/interceptors',
        VIEWS: 'app/views', // views directory
        CONTROLLERS: 'app/controllers', // controllers directory
        STATIC: "app/static", // static content directory
        RESOURCES: ".resources",
        DEFAULT_THEME: "default"
    },
    MYSQL: {
        HOST: "localhost",
        USER: 'root',
        PASSWORD: '',
        DATABASE: ''
    },
    STATIC: {
        CHARSET: "UTF-8",
        COMPACT_RESOURCES: false,
        COMPRESS_RESOURCES: false,
        CACHE_AGE: 645600,
        CACHE_RESOURCES: false,
        CACHE_HTML: false,
        COMPRESS_HTML: false,
        COMPRESS_HTML_EXCEPTION: [],
        CONTENT_TYPE: {
            PDF: 'application/pdf',
            CSS: 'text/css',
            JS: 'text/javascript',
            HTML: 'text/html',
            TXT: 'text/plain',
            GIF: 'image/gif',
            PNG: 'image/png',
            JPEG: 'image/jpeg',
            JPG: 'image/jpg'
        },
        RESOURCES: {
        }
    },
    ERROR: {
        REDIRECT: {
            //all:"/error/{code}",
            //404: "/error/404"
        }
    }
};
