// elegant-config/index.js
module.exports = {
    SERVER: {
        PORT: 8000,
        POST_MAX_SIZE: 40, //MB
        UPLOAD_MAX_FILE_SIZE: 40, //MB , less than or equal POST_MAX_SIZE
    },
    DIR: {
        PROJECT: __dirname, // project directory
        SESSION: __dirname + "/.session-data", // session data
        INTERCEPTORS: 'app/interceptors',
        VIEWS: 'app/views', // views directory
        CONTROLLERS: 'app/controllers', // controllers directory
        STATIC: "app/static" // static content directory
    },
    DEFAULT_THEME: "default",
    DATA_SOURCE: {
        SOURCE: "MYSQL",
        MYSQL: {
            host: "localhost",
            user: 'root',
            password: '',
            database: ''
        }
    },
    STATIC: {
        HEADERS: {
            pdf: 'application/pdf',
            css: 'text/css',
            js: 'text/javascript',
            html: 'text/html',
            txt: 'text/plain',
            jpg: 'image/jpeg',
            gif: 'image/gif',
            png: 'image/png',
            jpeg: 'image/jpeg',
            jpg: 'image/jpg'
        }
    },
    ERROR: {
        REDIRECT: {
            //all:"/error/{code}",
            //404: "/error/404"
        }
    }
};
