// elegant-controller/index.js
var trim = require("trimmer");
var Sync = require("elegant-sync");
var clone = require("clone");
var controllers = [];

function Controller(path, method) {
    this._sync = new Sync(this);
    this._execute;

    this.complete;
    this._exports = [];
    this.id = controllers.length - 1;
    this.path = trim(path, "/");
    this.method = method != undefined ? method.toLowerCase() : "get";
    controllers[this.path] = this;
}
// class methods
Controller.prototype.execute = function (fun) {
    this._execute = fun;
    return this;
};
Controller.prototype.start = function () {
    var data = this._execute();
    this._sync.start(data);
};

Controller.prototype.$ = function (fun) {
    return this._sync.$(fun, arguments);
};

Controller.prototype.ready = function (fun) {
    return this._sync.ready(fun);
};

Controller.prototype.export = function (name, val) {
    return this._exports[name] = val;
};

// export the class
var controller = module.exports = Controller;

// get all controllers
controller.all = function () {
    return clone(controllers);
};
