// elegant-controller/index.js
var trim = require("trimmer");
var Sync = require("elegant-sync");
var clone = require("clone");
var controllers = [];

function Controller(path) {
    this._sync = new Sync(this);
    this._execute;
    this.complete;
    this.method;
    this.id = controllers.length - 1;
    this.path = trim(path, "/");
    controllers.push(this);
}


Controller.prototype.get = function (fun) {
    this._execute = fun;
    this.method = "get";
    return this;
};

Controller.prototype.post = function (fun) {
    this._execute = fun;
    this.method = "post";
    return this;
};

Controller.prototype.start = function () {
    this._sync.start();
};

Controller.prototype.$ = function (fun) {
    return this._sync.$(fun, arguments);
};

Controller.prototype.ready = function (fun) {
    return this._sync.ready(fun);
};
Controller.prototype.error = function (fun) {
    return this._sync.error(fun);
};

// export the class
var controller = module.exports = Controller;

// get all controllers
controller.all = function () {
    return clone(controllers);
};
